### PR TITLE
Sharding functionality 

### PR DESCRIPTION
**Pull request by jbornemann**
This PR builds off of https://github.com/sclorg/mongodb-container/pull/264 to add Sharding functionality to the image. 

Some things changed:

- mongodb.conf template moved to /usr/share/container-scripts/mongodb/template folder. mongos.conf template created.
- mongos initialization routines added under /usr/share/container-scripts/mongodb/db/mongos
- run-mongos entrypoint added under /usr/bin

Other changes:

- mongo-is-ready script added to /usr/bin for OpenShift readiness probe checks. 

As noted in 264, : Just a note, https://github.com/jbornemann/mongodb-on-os I plan on submitting to container-quickstarts as an example of how all of these run modes fit together on OpenShift in a sharded Mongo environment.

We should probably ignore this PR until https://github.com/sclorg/mongodb-container/pull/264 is in.

### CODE REVIEW

**Comment by jbornemann** [comment]
@omron93

File: latest/root/usr/share/container-scripts/mongodb/README.md [hunk 1]
```diff
@@ -61,6 +61,21 @@ Optionally you can provide settings for user with 'readWrite' role.
...
+| :------------------------  | ---------------------------------| ---------|
+|  `MONGODB_KEYFILE_VALUE`   | Specific replica set secret      | -        |
+|  `MONGODB_REPLICA_NAME`    | Specific replica set name        | -        |
+|  `MONGODB_SERVICE_NAME`    | Specific replica set service     | mongodb  |
+
+Running image as Configuration Server
+|    Variable Name           |    Description                   |    Value |
+| :------------------------  | ---------------------------------| ---------|
+|    `CONFIG_REPLSET_NAME`   | Replica Set name of the config server  | |
+|   `CONFIG_REPLSET_SERVER`  | A single member URI of the config server replica set | |
```
Comment (omron93) [inline latest/root/usr/share/container-scripts/mongodb/README.md:76]: Why not to use OpenShift service for this? (same as used in replicaset scripts)
And leave less configuration to users.

File: latest/root/usr/share/container-scripts/mongodb/README.md [hunk 2]
```diff
@@ -61,6 +61,21 @@ Optionally you can provide settings for user with 'readWrite' role.
...
+|   `REPLSET_NAMES`          | Comma delimited names of replica sets apart of shard cluster | |
+|   `REPLSET_SERVERS`        | Comma delimited members of each replica set in shard cluster. One from each replica set, in order | |
```
Comment (omron93) [inline latest/root/usr/share/container-scripts/mongodb/README.md:78]: Why not to specify only names of services used by shard replicasets? (and use dynamic orchestration provided by OpenShift)

**Review by omron93 — COMMENTED** [review]
Only brief review. And waiting for adding functionality for configsvr - #264

**Comment by omron93** [comment]
@jbornemann I see that in mentioned example repo you have separate templates for configsvr, mongos and shard replicasets.
Will someone use some part separately? Or users will create them all together and then would be better to have on big template which will setup working sharded cluster? (or are there some guidelines that template should contain only one service, DC,... or something similar?)

File: latest/root/usr/share/container-scripts/mongodb/README.md (hunk 2 above)
Comment (jbornemann) [reply latest/root/usr/share/container-scripts/mongodb/README.md:78]: Ah, a good point. I think at the time I was hoping to decouple the image runtime as much as possible from OpenShift - In current implementation, I'm not sure how decoupled it actually is aside from this.

**Comment by jbornemann** [comment]
@omron93 A good question. :-)  Yes, it's a question of sharded vs non-sharded data; and which components are managed by OpenShift. Sometimes data won't be sharded - so no config server or mongos. Sometimes the mongos and/or config server will be managed outside of OpenShift - i.e bursting replicas into the cloud, in a hybrid cloud situation.

**Comment by ghost** [comment]
Can one of the admins verify this patch?

**Comment by hhorak** [comment]
mongodb container is not maintained any more in this org. closing.

**Pull request closed without merging**